ingest-app/src/cmr/ingest/validation/validation.clj

```diff
@@ -94,16 +94,41 @@
      :ISOTopicCategories (match-kms-keywords-validation
                           kms-index :iso-topic-categories msg/iso-topic-category-not-matches-kms-keywords)}))
 
+(defn- log-full-error-message
+  "Log entire validation error message so that information is not lost."
+  [error]
+  (info (str "UMM Validation error. Full message: " error)))
+
+(defn- create-user-friendly-error-message
+  "Remove non user-friendly things like regexes from error message to be returned.
+   The original message should be logged."
+  [error]
+  (let [friendly-error (string/split error #";")]
+    (if (>= (count friendly-error) 1)
+      (do
+       (log-full-error-message error)
+       (first friendly-error))
+      error)))
+
+(defn- sanitize-error-messages
+  "Remove nasty error messages from the collection to be returned.
+   Group errors by path and ensure that end-users don't have to see regexes. The horror!"
+  [error-messages]
+  (->> error-messages
+       (map create-user-friendly-error-message)
+       (remove #(re-find #"(ECMA|regex)" %))))
```

Comment:
I'm not understanding the changes. Why are we throwing away any part of a message after a semi-colon, and then it looks like we're also throwing away any error message that still has ECMA in it?

This seems like it is going to have some bad side effects of not returning errors that we do want to return. I think the fix needs to be much more targeted to recognize a regex message and remove that part of the message from the overall error.

Reply:
The error message in MMT as I saw it was semi-colon separated, and wanted to account for that as well as for when the messages were separated into `required: x actual: y`. This felt like the simplest way to decide whether or not to truncate or drop a message entirely

Reply:
The MMT message was created by this function:

```
(defn- create-success-status-message
  "If there are warnings, create a status string with warnings, otherwise status
  is nil"
  [warnings]
  (when (seq warnings)
    (str "Collection was updated successfully, but translating the collection "
         "to UMM-C had the following issues: "
         (string/join "; " warnings))))
```
It puts a semi-colon between each individual warning. So the semi-colon was not part of the original message in this case.

The warning message we wanted to remove from the example was:
/Platforms/1/LongName ECMA 262 regex "[\w\-&'()\[\]/."#$%\^@!*+=,][\w-&'”)

in this case I think the best thing to do would be to change the error message to:
/Platforms/1/LongName is in an invalid format rather than dropping it entirely. So it failed two validations, too short and invalid format.

We could probably do this with a string substitution of s/ECMA.*/is in an invalid format.

Reply:
I forgot to clarify:

The reason I don't want to drop it entirely is there may be cases where the only error returned is a regex error, so we definitely want to let them know which field was in a format that is considered invalid by the schema.

Reply:
Ohhhh okay I follow now. Thanks for clearing that up